third_party/amd/lib/TritonAMDGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -774,27 +774,37 @@ struct AtomicRMWOpConversion
     // vec = 1, numElements = 1 for scalar
     auto vec = getVectorSize(ptr);
     int numElems = 1;
+    bool forceF16packing = false;
+    Type packF16Ty = vec_ty(valueElemTy, 2);
     // tensor
     if (tensorTy) {
       auto valTy = cast<RankedTensorType>(val.getType());
-      Type elTy = valTy.getElementType();
-      vec = std::min<unsigned>(vec, llvm::isa<FloatType>(elTy) &&
-                                            elTy.getIntOrFloatBitWidth() == 16
-                                        ? 2
-                                        : 1);
+      vec = std::min<unsigned>(vec,
+                               llvm::isa<FloatType>(valueElemTy) &&
+                                       valueElemTy.getIntOrFloatBitWidth() == 16
+                                   ? 2
+                                   : 1);
+      // Force F16 packing in the case it couldn't be already packed, ISA
+      // supports global atomics packed instructions
+      forceF16packing = targetInfo.getISAFamily() == AMD::ISAFamily::CDNA3 &&
+                        vec == 1 && llvm::isa<FloatType>(valueElemTy) &&
+                        valueElemTy.getIntOrFloatBitWidth() == 16;
```

Comment:
I think we should specifically check `valueElemTy.isF16() || valueElemTyp.isBF16()`? That's the only two variants with hardware packed ISA right?